packages/dev/core/src/Misc/HighDynamicRange/panoramaToCubemap.ts

```diff
@@ -120,36 +120,45 @@ export class PanoramaToCubeMapTools {
         };
     }
 
-    private static CreateCubemapTexture(texSize: number, faceData: Vector3[], float32Array: Float32Array, inputWidth: number, inputHeight: number) {
+    private static CreateCubemapTexture(texSize: number, faceData: Vector3[], float32Array: Float32Array, inputWidth: number, inputHeight: number, supersample = false) {
         const buffer = new ArrayBuffer(texSize * texSize * 4 * 3);
         const textureArray = new Float32Array(buffer);
 
-        const rotDX1 = faceData[1].subtract(faceData[0]).scale(1 / texSize);
-        const rotDX2 = faceData[3].subtract(faceData[2]).scale(1 / texSize);
+        // If supersampling, determine number of samples needed when source texture width is divided for 4 cube faces
+        const samples = supersample ? Math.max(1, Math.round(inputWidth / 4 / texSize)) : 1;

```

Comment:
Why divide by 4? This means that we won't have supersampling as long as `inputWidth <= 4*texSize` (for eg: inputWidth=512 and texSize=128).